src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -47,68 +45,91 @@ var kubeconfigCmd = &cobra.Command{
 	Long:  `This command configures the Kubernetes configuration inside a WSL2 distribution.`,
 	Args:  cobra.ExactArgs(0),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		winConfigPath := kubeconfigViper.GetString("kubeconfig")
-		linuxConfigDir := filepath.Join(homedir.HomeDir(), ".kube")
-		linuxConfigPath := filepath.Join(linuxConfigDir, "config")
+		configPath := kubeconfigViper.GetString("kubeconfig")
 		enable := kubeconfigViper.GetBool("enable")
 
-		if winConfigPath == "" {
-			//lint:ignore ST1005 The capitalization is for a proper noun.
+		if configPath == "" {
 			return errors.New("Windows kubeconfig not supplied")
 		}
 
-		// Backwards compatibility: if the Linux config is a symlink to the Windows
-		// kubeconfig, unlink it.  This avoids issues where we clobber the Windows
-		// kubeconfig by accident.
-		if _, err := os.Readlink(linuxConfigPath); err == nil {
-			linuxInfo, err1 := os.Stat(linuxConfigPath)
-			windowsInfo, err2 := os.Stat(winConfigPath)
-			if err1 == nil && err2 == nil && os.SameFile(linuxInfo, windowsInfo) {
-				if err = os.Remove(linuxConfigPath); err != nil {
-					return fmt.Errorf("failed to remove kubeconfig symlink: %w", err)
-				}
-			}
-		}
-
-		if !enable {
-			return nil
+		_, err := os.Stat(configPath)
+		if err != nil {
+			return fmt.Errorf("could not open Windows kubeconfig: %w", err)
 		}
-
 		cmd.SilenceUsage = true
 
-		winConfig, err := readKubeConfig(winConfigPath)
-		if err != nil {
-			return err
+		configDir := path.Join(homedir.HomeDir(), ".kube")
+		linkPath := path.Join(configDir, "config")
+		manualSymlink, symlinkErr := requireManualSymlink(linkPath)
```

Comment:
Hmm, would calling `manualSymlink` something like `foreignKubeconfig` be better?